File: codex/cpp/Strings/Padding.cpp
// cpp, strings, padding strings, iomanip, setw, setiosflags

#include <iomanip>
#include <iostream>

std::string str {"Hello"};

std::cout << std::setw(7) << str << std::endl;  // "  Hello"
std::cout << std::setiosflags(std::ios::left)
     << std::setw(7) << str << std::endl;       // "Hello  "

File: codex/r/Strings/Padding.R
# r, strings, padding strings, str_pad

str <- "Hello"
str_pad(str, 7)                 # "  Hello"
str_pad(str, 7, side="left")    # "  Hello"
str_pad(str, 7, side="right")   # "Hello  "
str_pad(str, 10, side="both")   # "  Hello   "
str_pad(str, 10, pad='_')       # "_____Hello"
